pkg/sql/logictest/testdata/logic_test/collatedstring

```diff
@@ -368,3 +368,18 @@ CREATE TABLE t46570(c0 BOOL, c1 STRING COLLATE en);
 CREATE INDEX ON t46570(rowid, c1 DESC);
 INSERT INTO t46570(c1, rowid) VALUES('' COLLATE en, 0);
 UPSERT INTO t46570(rowid) VALUES (0), (1)
+
+subtest regression_50015
+
+statement ok
```

Comment:
can you add a comment about what this test is doing + what does the output mean?